### ISSUE TITLE
Example 1 crashes when throwing random boxes

### ISSUE THREAD

**Issue by raffaello-camoriano** [xbpeng/DeepMimic#58]
When trying to run Example 1 from https://github.com/xbpeng/DeepMimic#how-to-use on my setup, the simulation starts and runs properly (including with external forces input via mouse), but crashes as soon as 'x' is pressed to throw random boxes to the character.
I tried this multiple times and it always crashes.

**Host system info**
- OS: Ubuntu 18.04 (bionic)
- Kernel: 4.15.0-47-generic (#50-Ubuntu SMP Wed Mar 13 10:44:52 UTC 2019)
- GCC Version: 7 (x86_64-linux-gnu)

**Details**
```
kammo@kammo-XPS-15-9560:~/repos/DeepMimic$ python DeepMimic.py --arg_file args/run_humanoid3d_spinkick_args.txt

WARNING: The TensorFlow contrib module will not be included in TensorFlow 2.0.
For more information, please see:
  * https://github.com/tensorflow/community/blob/master/rfcs/20180907-contrib-sunset.md
  * https://github.com/tensorflow/addons
If you depend on functionality not listed there, please file an issue.

Renderer: GeForce GTX 1050/PCIe/SSE2
OpenGL version supported 3.2.0 NVIDIA 415.27
Compiling shader: data/shaders/Mesh_VS.glsl
Compiling shader: data/shaders/VertColor_PS.glsl
Compiling shader: data/shaders/FullScreenQuad_VS.glsl
Compiling shader: data/shaders/DownSample_PS.glsl
Compiling shader: data/shaders/Mesh_VS.glsl
Compiling shader: data/shaders/DownSample_PS.glsl

Agent Registry
Num Agents: 1
Agent 0: ct_pd

Compiling shader: data/shaders/Mesh_VS.glsl
Compiling shader: data/shaders/Depth_PS.glsl
Compiling shader: data/shaders/Mesh_VS.glsl
Compiling shader: data/shaders/Lighting_Lambert_PS.glsl
Loaded scene: Imitate

Num Agents: 1
Agent 0: data/agents/ct_agent_humanoid_ppo.txt
2019-05-16 23:19:01.387353: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2019-05-16 23:19:01.408368: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2808000000 Hz
2019-05-16 23:19:01.408836: I tensorflow/compiler/xla/service/service.cc:150] XLA service 0x3e5da10 executing computations on platform Host. Devices:
2019-05-16 23:19:01.408852: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (0): <undefined>, <undefined>
WARNING:tensorflow:From /home/kammo/.local/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
WARNING:tensorflow:From /home/kammo/repos/DeepMimic/learning/tf_util.py:34: dense (from tensorflow.python.layers.core) is deprecated and will be removed in a future version.
Instructions for updating:
Use keras.layers.dense instead.
Built actor net: fc_2layers_1024units
Built critic net: fc_2layers_1024units
WARNING:tensorflow:From /home/kammo/repos/DeepMimic/learning/tf_util.py:53: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
WARNING:tensorflow:From /home/kammo/.local/lib/python3.6/site-packages/tensorflow/python/ops/math_ops.py:3066: to_int32 (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
{
"ID": 0,
 "Type": "PPO",
 "ActionSpace": "Continuous",
 "StateDim": 197,
 "GoalDim": 0,
 "ActionDim": 36
}
WARNING:tensorflow:From /home/kammo/.local/lib/python3.6/site-packages/tensorflow/python/training/saver.py:1266: checkpoint_exists (from tensorflow.python.training.checkpoint_management) is deprecated and will be removed in a future version.
Instructions for updating:
Use standard file APIs to check for files with this prefix.
Model loaded from: data/policies/humanoid3d/humanoid3d_spinkick.ckpt

python3: util/MathUtil.cpp:175: static tMatrix cMathUtil::RotateMat(const tVector &, double): Assertion `std::abs(axis.squaredNorm() - 1) < 0.0001' failed.
[kammo-XPS-15-9560:01642] *** Process received signal ***
[kammo-XPS-15-9560:01642] Signal: Aborted (6)
[kammo-XPS-15-9560:01642] Signal code:  (-6)
[kammo-XPS-15-9560:01642] [ 0] /lib/x86_64-linux-gnu/libc.so.6(+0x3ef20)[0x7fe71c3b4f20]
[kammo-XPS-15-9560:01642] [ 1] /lib/x86_64-linux-gnu/libc.so.6(gsignal+0xc7)[0x7fe71c3b4e97]
[kammo-XPS-15-9560:01642] [ 2] /lib/x86_64-linux-gnu/libc.so.6(abort+0x141)[0x7fe71c3b6801]
[kammo-XPS-15-9560:01642] [ 3] /lib/x86_64-linux-gnu/libc.so.6(+0x3039a)[0x7fe71c3a639a]
[kammo-XPS-15-9560:01642] [ 4] /lib/x86_64-linux-gnu/libc.so.6(+0x30412)[0x7fe71c3a6412]
[kammo-XPS-15-9560:01642] [ 5] /home/kammo/repos/DeepMimic/DeepMimicCore/_DeepMimicCore.so(_ZN9cMathUtil9RotateMatERKN5Eigen6MatrixIdLi4ELi1ELi0ELi4ELi1EEEd+0x1b3)[0x7fe6f5976183]
[kammo-XPS-15-9560:01642] [ 6] /home/kammo/repos/DeepMimic/DeepMimicCore/_DeepMimicCore.so(_ZN12cMatrixStack6RotateEdRKN5Eigen6MatrixIdLi4ELi1ELi0ELi4ELi1EEE+0x1d)[0x7fe6f5a4cefd]
[kammo-XPS-15-9560:01642] [ 7] /home/kammo/repos/DeepMimic/DeepMimicCore/_DeepMimicCore.so(_ZN8cDrawObj7DrawBoxEPK7cSimObjRKN5Eigen6MatrixIdLi4ELi1ELi0ELi4ELi1EEES7_N9cDrawUtil9eDrawModeE+0x86)[0x7fe6f5a43006]
[kammo-XPS-15-9560:01642] [ 8] /home/kammo/repos/DeepMimic/DeepMimicCore/_DeepMimicCore.so(_ZN8cDrawObj4DrawEPK7cSimObjN9cDrawUtil9eDrawModeE+0x62)[0x7fe6f5a42b62]
[kammo-XPS-15-9560:01642] [ 9] /home/kammo/repos/DeepMimic/DeepMimicCore/_DeepMimicCore.so(_ZNK17cDrawSceneSimChar8DrawObjsEv+0x65)[0x7fe6f5a8c445]
[kammo-XPS-15-9560:01642] [10] /home/kammo/repos/DeepMimic/DeepMimicCore/_DeepMimicCore.so(_ZN10cDrawScene12DoShadowPassEv+0x120)[0x7fe6f5a9fcc0]
[kammo-XPS-15-9560:01642] [11] /home/kammo/repos/DeepMimic/DeepMimicCore/_DeepMimicCore.so(_ZN10cDrawScene9DrawSceneEv+0xd)[0x7fe6f5a9ea7d]
[kammo-XPS-15-9560:01642] [12] /home/kammo/repos/DeepMimic/DeepMimicCore/_DeepMimicCore.so(_ZTv0_n72_N10cDrawScene4DrawEv+0x3a)[0x7fe6f5a9dcda]
[kammo-XPS-15-9560:01642] [13] /home/kammo/repos/DeepMimic/DeepMimicCore/_DeepMimicCore.so(_ZN14cDeepMimicCore4DrawEv+0x22)[0x7fe6f596a182]
[kammo-XPS-15-9560:01642] [14] /home/kammo/repos/DeepMimic/DeepMimicCore/_DeepMimicCore.so(+0x1f7649)[0x7fe6f5ab6649]
[kammo-XPS-15-9560:01642] [15] python3[0x502d6f]
[kammo-XPS-15-9560:01642] [16] python3(_PyEval_EvalFrameDefault+0x449)[0x506859]
[kammo-XPS-15-9560:01642] [17] python3[0x502209]
[kammo-XPS-15-9560:01642] [18] python3[0x502f3d]
[kammo-XPS-15-9560:01642] [19] python3(_PyEval_EvalFrameDefault+0x449)[0x506859]
[kammo-XPS-15-9560:01642] [20] python3[0x502209]
[kammo-XPS-15-9560:01642] [21] python3[0x502f3d]
[kammo-XPS-15-9560:01642] [22] python3(_PyEval_EvalFrameDefault+0x449)[0x506859]
[kammo-XPS-15-9560:01642] [23] python3[0x504c28]
[kammo-XPS-15-9560:01642] [24] python3[0x58644b]
[kammo-XPS-15-9560:01642] [25] python3(PyObject_Call+0x3e)[0x59ebbe]
[kammo-XPS-15-9560:01642] [26] /usr/lib/python3.6/lib-dynload/_ctypes.cpython-36m-x86_64-linux-gnu.so(+0xbb74)[0x7fe7068d4b74]
[kammo-XPS-15-9560:01642] [27] /usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_closure_unix64_inner+0x18f)[0x7fe7066c6b4f]
[kammo-XPS-15-9560:01642] [28] /usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_closure_unix64+0x46)[0x7fe7066c6f16]
[kammo-XPS-15-9560:01642] [29] /usr/local/lib/libglut.so.3(fghRedrawWindow+0x50)[0x7fe703348237]
[kammo-XPS-15-9560:01642] *** End of error message ***
Aborted (core dumped)
```

**Comment by xbpeng** [comment]
it's hard to debug with the errors from the wrapper. To dig further into this, you might need to run the app in c++ and see what triggers that crash. Main.cpp is the source for running the app in c++, maybe give that a try? To build the app run
make
instead of
make python
Unfortunately, I haven't ran the c++ app much in linux, so not sure if it will work properly.

**Comment by zhaolongkzz** [comment]
@raffaello-camoriano, hello, I found the same question and get the wrong message below.
details
```bash
Built actor net: fc_2layers_1024units
Built critic net: fc_2layers_1024units
{
"ID": 0,
 "Type": "PPO",
 "ActionSpace": "Continuous",
 "StateDim": 197,
 "GoalDim": 0,
 "ActionDim": 36
}
Model loaded from: data/policies/humanoid3d/humanoid3d_walk.ckpt

python: util/MathUtil.cpp:175: static tMatrix cMathUtil::RotateMat(const tVector &, double): Assertion `std::abs(axis.squaredNorm() - 1) < 0.0001' failed.
Aborted
```
And I adjust the equation `std::abs(axis.squaredNorm() - 1) < 0.1`, from 0.0001 to 0.1, then it never show wrong message again! (I think it's the problem of blocks size)
Hope it can help you!

**Comment by raffaello-camoriano** [comment]
Thank you very much @zhaolongkzz, it did work!

**Comment by rongduo** [comment]
Thanks a lot @zhaolongkzz , I also met this problem and I solved it according to your solution.

**Comment by tommysugiarto** [comment]
Hi, I also face the same problem, could you help me where is / how to locate the util/MatUtil.cpp so that i can modified the code?

Thanks
